fuse_optimizers/src/fixed_lag_smoother.cpp

```diff
@@ -332,13 +333,28 @@ void FixedLagSmoother::processQueue(fuse_core::Transaction& transaction)
     }
   }
 
+  // Use the most recent transaction time as the current time
+  const auto current_time =
```

Comment:
Having to do a linear search here is annoying. *If* the pending queue was sorted by max time, then this would just be the end of the pending queue. And the `(min_stamp < lag_expiration)` filtering logic will still purge out transactions involving old stamps.